torch/csrc/jit/codegen/cuda/predicate_compute.cpp

```diff
@@ -106,6 +107,82 @@ std::vector<kir::Bool*> PredicateCompute::computePredicates(
   return preds;
 }
 
+namespace {
+
+//! Analyze whether IterDomain can be statically determined to be safe
+//! without bounds-checking predicates.
+class IterationDomainAnalysis : private ExpressionEvaluator {
```

Comment:
we're using inheritance when composability would be a better choice - another sign that changing the EE interface is not  neccesary